SharedSource/UnitTests/Reflection/TypeExtensionsTest.cs

```diff
@@ -0,0 +1,170 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+
+using System;
+using System.Reflection;
+using NUnit.Framework;
+using Remotion.Reflection;
+
+#nullable enable
+// ReSharper disable once CheckNamespace
+namespace Remotion.UnitTests.Reflection
+{
+  [TestFixture]
+  public class TypeExtensionsTest
+  {
+    [Test]
+    public void GetAssemblyQualifiedNameSafe_SystemType_AssemblyQualifiedName ()
+    {
+      var typeArgument = typeof (string);
+      const string expectedAssemblyQualifiedNamePrefix = "System.String,";
+
+      Assert.That (typeArgument.AssemblyQualifiedName, Is.Not.Null);
+      Assert.That (typeArgument.GetAssemblyQualifiedNameSafe(), Does.StartWith (expectedAssemblyQualifiedNamePrefix));
```

Comment:
>  I opted against doing so because I did not want to "mirror" the implementation of GetAssemblyQualifiedNameSafe too much, but rather test based on the understanding of what an assembly qualified name is.
Yes, that's the reasoning I usually employ in such cases as well. Here, it's just that not the actual intended case. We don't want to emulate an AQN but rather show that we return THE AssemblyQualifiedName for the type used in the test, as opposted to the FullName or the Name. So we don't care about the details, and as long as we can assume it as a given that AQN, FullName and Name will be different, there's no need to be more explicit when ensuring that the correct property is used. If we could mock the type, we could be even more explicit, of course.

> So, if I understand correctly, it's fine to leave it as it is?
Yup.